test/extended/router/metrics.go

```diff
@@ -186,7 +186,73 @@ var _ = g.Describe("[sig-network][Feature:Router]", func() {
 			o.Expect(findGaugesWithLabels(metrics["haproxy_backend_up"], routeLabels)).To(o.Equal([]float64{1}))
 			o.Expect(findGaugesWithLabels(metrics["haproxy_server_bytes_in_total"], serverLabels)).To(o.ConsistOf(o.BeNumerically(">=", 0), o.BeNumerically(">=", 0)))
 			o.Expect(findGaugesWithLabels(metrics["haproxy_server_bytes_out_total"], serverLabels)).To(o.ConsistOf(o.BeNumerically(">=", 0), o.BeNumerically(">=", 0)))
-			o.Expect(findGaugesWithLabels(metrics["haproxy_server_max_sessions"], serverLabels)).To(o.ConsistOf(o.BeNumerically(">", 0), o.BeNumerically(">", 0)))
+
+			// haxproxy_server_max_sessions needs to be
+			// greater than zero over two successive
+			// scrapes once the first scrape shows an
+			// increase in total connections.
+			err = wait.PollImmediate(2*time.Second, 5*time.Minute, func() (bool, error) {
+				refreshMetrics := func() (map[string]*dto.MetricFamily, error) {
+					startTime := time.Now()
+					results, err := getBearerTokenURLViaPod(ns, execPodName, fmt.Sprintf("http://%s:%d/metrics", host, metricsPort), bearerToken)
+					if err != nil {
+						return nil, err
+					}
+					metrics, err := p.TextToMetricFamilies(bytes.NewBufferString(results))
+					e2e.Logf("metrics refresh completed in %s", time.Now().Sub(startTime).Round(time.Second).String())
+					return metrics, err
+				}
+
+				maxSessionsInvariant := func(v []float64) bool {
+					return len(v) >= 2 && v[0] > 0 && v[1] > 0
+				}
+
+				updatedMetrics, err := refreshMetrics()
+				if err != nil {
+					e2e.Logf("refresh metrics error: %v", err)
+					return false, err
+				}
+
+				if len(findGaugesWithLabels(updatedMetrics["haproxy_server_up"], serverLabels)) < 2 {
+					e2e.Logf("haproxy_server_up: %v", findGaugesWithLabels(updatedMetrics["haproxy_server_up"], serverLabels))
+					g.By("sending traffic to a weighted route")
+					return false, expectRouteStatusCodeRepeatedExec(ns, execPodName, fmt.Sprintf("http://%s", host), "weighted.metrics.example.com", http.StatusOK, times, proxyProtocol)
+				}
+
+				if findGaugesWithLabels(updatedMetrics["haproxy_backend_connections_total"], routeLabels)[0] < float64(times) {
```

Comment:
Are the other values in this slice not important? Would it be worth checking the entire slice?